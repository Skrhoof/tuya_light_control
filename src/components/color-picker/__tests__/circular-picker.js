import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CircularPicker from '../circular-picker';

describe('CircularPicker Component', () => {
  it('basic render', () => {
    const component = renderer.create(<CircularPicker />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render picker', () => {
    const component = renderer.create(
      <CircularPicker
        degree={0}
        radius={115}
        frontStrokeColor="blue"
        strokeColor="red"
        strokeWidth={1}
        startDegree={30}
        endDegree={330}
        disabled={false}
        TrackComponent={null}
        onValueChange={() => { }}
        onComplete={() => { }}
      />
    );
    const instance = component.root.instance;
    instance._thumbRef.setNativeProps = jest.fn()
    instance.updateThumbStyle({ color: 'red' });
    expect(component.toJSON()).toMatchSnapshot();
  })

  it('should adjust svg coords properly', () => {
    const component = renderer.create(<CircularPicker />);
    const instance = component.root.instance;
    expect(instance.adjustSvgCoordinates(30, 30, Math.PI)).toEqual(Math.PI)
    expect(instance.adjustSvgCoordinates(-30, 30, Math.PI)).toEqual(Math.PI)
    expect(instance.adjustSvgCoordinates(-30, -30, -Math.PI)).toEqual(Math.PI)
    expect(instance.adjustSvgCoordinates(30, -30, -Math.PI)).toEqual(Math.PI)
    expect(instance.adjustSvgCoordinates(0, 30, -Math.PI)).toEqual(Math.PI * 3 / 2)
    expect(instance.adjustSvgCoordinates(0, -30, -Math.PI)).toEqual(Math.PI / 2)
    expect(instance.adjustSvgCoordinates(0, 0, -Math.PI)).toEqual(0)
  });

  it('should not set panresponder', () => {
    const component = renderer.create(<CircularPicker />);
    const instance = component.root.instance;
    expect(instance.shouldSetResponder()).toEqual(true)
  })

  it('should move correctly', () => {
    const component = renderer.create(<CircularPicker />);
    const instance = component.root.instance;
    instance._thumbRef.setNativeProps = jest.fn()
    instance._moveTo(100, 100);
    expect(instance._thumbRef.setNativeProps).toHaveBeenCalled();

    const x = instance._centerPoint + instance._thumbRadius;
    instance._moveTo(x, x + 100);
    expect(instance._thumbRef.setNativeProps).toHaveBeenCalled();
  })

  it('should gesture respond properly', () => {
    const component = renderer.create(<CircularPicker />);
    const instance = component.root.instance;
    instance._thumbRef.setNativeProps = jest.fn();

    instance._handleGrant({ nativeEvent: { locationX: 100, locationY: 100 } });
    expect(instance.xRelativeOriginStart).toEqual(100);
    expect(instance.yRelativeOriginStart).toEqual(100);

    instance._handleMove({}, { dx: 100, dy: 100 });
    expect(instance._thumbRef.setNativeProps).toHaveBeenCalled();

    instance._handleRelease({}, { dx: 150, dy: 150 })
    expect(instance._thumbRef.setNativeProps).toHaveBeenCalled();
  })

  it('should component update property', () => {
    const component = renderer.create(<CircularPicker key={'some-key'}/>);
    const instance = component.root.instance;
    const radius = 1000;
    const strokeWidth = 1000;
    component.update(<CircularPicker key={'some-key'} radius={radius} strokeWidth={strokeWidth}/>)
    expect({
      r: instance._thumbRadius,
      w: instance._centerPoint
    }).toEqual({
      r: strokeWidth * 0.5 + 2,
      w: radius + strokeWidth * 0.5 + 2
    })
  });

  it('should render disabled picker', () => {
    const component = renderer.create(
      <CircularPicker
        degree={0}
        radius={115}
        frontStrokeColor="blue"
        strokeColor="red"
        strokeWidth={1}
        startDegree={30}
        endDegree={330}
        disabled={true}
        TrackComponent={null}
        onValueChange={() => { }}
        onComplete={() => { }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  })
});

