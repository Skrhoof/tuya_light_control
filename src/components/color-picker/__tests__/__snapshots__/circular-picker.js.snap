// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularPicker Component basic render 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  pointerEvents="box-only"
  style={
    Array [
      null,
      Object {
        "height": 274,
        "width": 274,
      },
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 274,
          "width": 274,
        },
      ]
    }
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    >
      <RNSVGCircle
        cx="137"
        cy="137"
        fill={
          Array [
            0,
            0,
            0,
            0,
            0,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
            "fill",
          ]
        }
        r="115"
        stroke={
          Array [
            0,
            1,
            0,
            0,
            1,
          ]
        }
        strokeDasharray={
          Array [
            602.1385919380438,
            120.42771838760872,
          ]
        }
        strokeDashoffset={-240.85543677521744}
        strokeLinecap={1}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={40}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Image
    source={1}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
          "position": "absolute",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                0,
              ],
            },
          ],
        },
        Object {
          "borderRadius": 22,
          "height": 44,
          "opacity": 1,
          "transform": Array [
            Object {
              "translate": Array [
                115.00000000000003,
                230,
              ],
            },
          ],
          "width": 44,
        },
        null,
      ]
    }
  />
</View>
`;

exports[`CircularPicker Component should render disabled picker 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  pointerEvents="box-only"
  style={
    Array [
      null,
      Object {
        "height": 235,
        "width": 235,
      },
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 235,
          "width": 235,
        },
      ]
    }
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    >
      <RNSVGCircle
        cx="117.5"
        cy="117.5"
        fill={
          Array [
            0,
            0,
            0,
            0,
            0,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
            "fill",
          ]
        }
        r="115"
        stroke={
          Array [
            0,
            1,
            0,
            0,
            1,
          ]
        }
        strokeDasharray={
          Array [
            602.1385919380438,
            120.42771838760872,
          ]
        }
        strokeDashoffset={-240.85543677521744}
        strokeLinecap={1}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
      />
      <RNSVGCircle
        cx="117.5"
        cy="117.5"
        fill={
          Array [
            0,
            0,
            0,
            0,
            0,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
            "fill",
          ]
        }
        r="115"
        stroke={
          Array [
            0,
            0,
            0,
            1,
            1,
          ]
        }
        strokeDasharray={
          Array [
            0,
            722.5663103256525,
          ]
        }
        strokeDashoffset={-240.85543677521744}
        strokeLinecap={1}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Image
    source={1}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
          "position": "absolute",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                0,
              ],
            },
          ],
        },
        Object {
          "borderRadius": 2.5,
          "height": 5,
          "opacity": 0,
          "transform": Array [
            Object {
              "translate": Array [
                115.00000000000003,
                230,
              ],
            },
          ],
          "width": 5,
        },
        null,
      ]
    }
  />
</View>
`;

exports[`CircularPicker Component should render picker 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  pointerEvents="box-only"
  style={
    Array [
      null,
      Object {
        "height": 235,
        "width": 235,
      },
    ]
  }
>
  <RNSVGSvgView
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 235,
          "width": 235,
        },
      ]
    }
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    >
      <RNSVGCircle
        cx="117.5"
        cy="117.5"
        fill={
          Array [
            0,
            0,
            0,
            0,
            0,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
            "fill",
          ]
        }
        r="115"
        stroke={
          Array [
            0,
            1,
            0,
            0,
            1,
          ]
        }
        strokeDasharray={
          Array [
            602.1385919380438,
            120.42771838760872,
          ]
        }
        strokeDashoffset={-240.85543677521744}
        strokeLinecap={1}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
      />
      <RNSVGCircle
        cx="117.5"
        cy="117.5"
        fill={
          Array [
            0,
            0,
            0,
            0,
            0,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
            "fill",
          ]
        }
        r="115"
        stroke={
          Array [
            0,
            0,
            0,
            1,
            1,
          ]
        }
        strokeDasharray={
          Array [
            0,
            722.5663103256525,
          ]
        }
        strokeDashoffset={-240.85543677521744}
        strokeLinecap={1}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Image
    source={1}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
          "position": "absolute",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                0,
              ],
            },
          ],
        },
        Object {
          "borderRadius": 2.5,
          "height": 5,
          "opacity": 1,
          "transform": Array [
            Object {
              "translate": Array [
                115.00000000000003,
                230,
              ],
            },
          ],
          "width": 5,
        },
        null,
      ]
    }
  />
</View>
`;
